### PR TITLE
add pagination for results

### PR DESCRIPTION
**Pull request by macjabeth**
This PR addresses issue #25. I'm open to making any changes, just lemme know!

### CODE REVIEW

**Comment by carlzoo** [comment]
Hi @macjabeth. Thanks for your PR and interest in the project. I think I should have been more clear that what I was looking for was a page that expanded rather than a left/right arrow. Nonetheless the logic looks correct.

File: src/components/Ticketmaster.js [hunk 1]
```diff
@@ -96,16 +96,33 @@ const ResultListMobile = Styled.ul`
...
+  button {
+    font-size: 1.2rem;
+  }
+`;
+
 const Ticketmaster = function() {
-  const dummy = JSON.parse('{ "_embedded" : { "events": [] }}');
+  const dummy = {
+    _embedded: { events: [] },
+    _links: {},
```
Comment (carlzoo) [inline src/components/Ticketmaster.js]: this variable is not used?

**Comment by macjabeth** [comment]
> Hi @macjabeth. Thanks for your PR and interest in the project. I think I should have been more clear that what I was looking for was a page that expanded rather than a left/right arrow. Nonetheless the logic looks correct.

Oh, oops! 😅 By "a page that expanded", do you mean like an infinite scroll kinda deal? Cuz I could take a go at that too!

File: src/components/Ticketmaster.js [hunk 2]
```diff
@@ -195,6 +218,16 @@ const Ticketmaster = function() {
             </li>
             ))}
     </ResultListMobile>
+    <Pagination role="navigation" aria-label="Pagination">
+      <ul>
+        <li>
+          <button onClick={prevPage} disabled={page === 0} aria-label="Next Page">&laquo;</button>
```
Comment (carlzoo) [inline src/components/Ticketmaster.js]: Sorry for not being clear but this should be one button which will expand the page vertically with the results.

File: src/components/Ticketmaster.js (hunk 1 above)
Comment (macjabeth) [reply src/components/Ticketmaster.js]: True. I was being explicit, but I could remove any unused variables therein if you like. 🙂

**Comment by carlzoo** [comment]
> 
> 
> > Hi @macjabeth. Thanks for your PR and interest in the project. I think I should have been more clear that what I was looking for was a page that expanded rather than a left/right arrow. Nonetheless the logic looks correct.
> 
> Oh, oops! sweat_smile By "a page that expanded", do you mean like an infinite scroll kinda deal? Cuz I could take a go at that too!

Infinite scroll is the right term.

**Comment by macjabeth** [comment]
> Infinite scroll is the right term.

@carlzoo Awesome. So just to be very clear, would you prefer a "Load more..." button that a user can click to load more results? Or would you like the results to be loaded automatically as they scroll down?

**Comment by carlzoo** [comment]
I think the button works. Can you also make a change to the appropriate item on README.md?

Something else which comes up is that the sidebar (for desktop) and navbar(for mobile) needs to be at a fixed position - but this is a separate issue and I'm not expecting it to be completed here. But if you can make this change this would be great.

**Comment by macjabeth** [comment]
@carlzoo Alrighty, lemme know what you think of those changes. 😊

**Comment by carlzoo** [comment]
The search results aren't returning correctly like the original version. Entering a new query shouldn't be showing portions of the original results. I believe the hook variables need to be cleared whenever a new search query is entered.

Also I think the search bar needs to be fixed position as well.

**Comment by macjabeth** [comment]
Thanks for pointing the bug out there - should be fixed now. On a side note, adding some tests to your code would probably be a good idea to ensure no errors like that pop up when changes are added.

**Comment by macjabeth** [comment]
In terms of UI, adding a fixed search bar would be a bit tricky as fixed elements don't respect the widths of their parent containers. As an alternative, I added a button for users to click which takes them back to the top of the page (and the search bar). Hope this will suffice. 🙂

**Review by conoroshea1996 — COMMENTED** [review]
This looks good and I think the button works well. I was playing with the fixed search-bar to try and make it work and like you have said its alittle tricky I mean you could do it but I think this works better.
Nicely done.

**Comment by carlzoo** [comment]
Looks good thanks for the PR.

**Pull request merged**